pasture-algorithms/src/normal_estimation.rs

```diff
@@ -190,48 +194,35 @@ fn is_finite(point: &Vector3<f64>) -> bool {
 
 /// let interleaved = points.into_iter().collect::<VectorBuffer>();
 
-/// let centroid = compute_centroid(&interleaved);
+/// let centroid = compute_centroid(&interleaved.view_attribute::<Vector3<f64>>(&POSITION_3D));
 ///
 /// ```
-pub fn compute_centroid<'a, T: BorrowedBuffer<'a>>(point_cloud: &'a T) -> Vector3<f64> {
-    if point_cloud.is_empty() {
-        panic!("The point cloud is empty!");
+pub fn compute_centroid<'a, 'b, T, F>(
+    attribute_view: &AttributeView<'a, 'b, T, Vector3<F>>
```

Comment:
`AttributeView` is `Copy`, so you can pass by value here. This prevents you from having to implement `IntoIterator` for `&AttributeView`